covalent/executor/utils/local_executor_utils.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+"""
+Helper functions for the local executor
+"""
+
+import io
+import multiprocessing as mp
+import traceback
+from contextlib import redirect_stderr, redirect_stdout
+from typing import Callable, Dict, List
+
+
+def local_wrapper(fn: Callable, args: List, kwargs: Dict, q: mp.Queue):
```

Comment:
I see there is code duplication here with `dask_wrapper` @cjao 

Does it maybe make sense for the AbtsractBaseExecutor to have a method that is in general mediating the actual execution on a function with the redirect stream logic and properly formatting the traceback? it may be possible more executors such as the Executor or AsyncCustomExecutor to potentially re-use this method

Reply:
Thanks for your feedback! This is actually subtly different from the `dask_wrapper` because of the multiprocessing Queue parameter. In effect, `local_wrapper` replicates the logic used by [ProcessPoolExecutor](https://github.com/python/cpython/blob/3.11/Lib/concurrent/futures/process.py#L255-L260) in concurrent.futures.

Reply:
Got it @cjao , given that these wrapper functions are still tied with a particular executor it may make sense to make them class methods of the particular executor they are tied to as they seem to share the same logical grouping. As for lines 35-41 however those are the lines which are repeated for dask/local and might be sensible to expect that those could also be used by other executors so may have sense to factor that into it's own function where the dask/local wrapper functions utilize that new method. Let me know your thoughts on this, thanks!

Reply:
This is a great suggestion. The wrappers used by local and dask are stored in a wrappers module under `covalent/executors/utils`.